### PR TITLE
Cleanup of e2e tests for consistency

### PR DESCRIPTION
**Pull request by PhilipGough**
Following on from the changes in #1280 this PR refactors the e2e tests with the following goals:

1. Use the recently added framework assertions to reduce boilerplate and make the tests more readable
1. Add some consistency across the e2e suite in regards to how the tests are written

@simonpasquier, @fpetkovski nothing urgent here but would appreciate a look when the time allows
 

* [ ] I added CHANGELOG entry for this change.
* [x] No user facing changes, so no entry in CHANGELOG was needed.

### CODE REVIEW

File: test/e2e/framework/helpers.go [hunk 1]
```diff
@@ -18,3 +22,41 @@ func (f *Framework) MustCreateOrUpdateConfigMap(t *testing.T, cm *v1.ConfigMap)
 		t.Fatalf("failed to create/update configmap - %s", err.Error())
 	}
 }
+
+// MustGetConfigMap or fail
```
Comment (fpetkovski) [inline test/e2e/framework/helpers.go:26]: I would maybe improve the godoc a bit with something like:
```suggestion
// MustGetConfigMap tries to get the `namespace/name` ConfigMap by polling it for 5 minutes. 
// It fails if the ConfigMap is not found within the polling period.
```

**Review by fpetkovski — COMMENTED** [review]
Great stuff, I left just a few minor suggestions.

File: test/e2e/framework/assertions.go [hunk 2]
```diff
@@ -368,6 +368,24 @@ func (f *Framework) AssertOperatorCondition(t *testing.T, conditionType configv1
 	}
 }
 
+func (f *Framework) AssertValueInClusterConfigMapEquals(key string, compareWith string) func(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (PhilipGough) [reply test/e2e/framework/assertions.go:371]: No real benefit, this format matches the style of the existing table driven tests. So it's either a matter of adding new assertions with this format or restructuring the format entirely. I'd be ok with either, but I don't personally see the benefit in a complete overhaul, wdyt?

[the remark this answers is not in the document]
Comment (fpetkovski) [reply test/e2e/framework/assertions.go:371]: Ack, this makes sense. We can have the assertions written in the same style as the scenarios.

**Review by arajkumar — APPROVED** [review]
overall looks good to me with a nit.

File: test/e2e/alertmanager_test.go [hunk 3]
```diff
@@ -119,7 +84,7 @@ func TestAlertmanagerKubeRbacProxy(t *testing.T) {
 	if err != nil {
 		t.Fatal(err)
 	}
-	defer cleanUp()
+	t.Cleanup(cleanUp)
```
[the remark this answers is not in the document]
Comment (PhilipGough) [reply test/e2e/alertmanager_test.go:87]: in this case it doesn't really matter, but the way `t.Cleanup` works allows us to embed it in test helpers and have resources cleaned up guaranteed after the subtests run, see https://github.com/openshift/cluster-monitoring-operator/pull/1378/files#diff-e4a57f3c1aaff059fed19689d2761dbea4caaddd364c3c034e9a7edb41e41b19R58 for example.

I modified here also just for consistency across tests, but in some cases, the functionality of Cleanup is desired

**Comment by PhilipGough** [comment]
/label docs-approved

No docs required updates e2e tests only

**Review by jan--f — COMMENTED** [review]
/lgtm

Nice work! `+335 −5,596` :heart:

**Pull request merged**